lib/galaxy/tools/cross_product_nested.xml

```diff
@@ -76,12 +76,69 @@
 Synopsis
 ========
 
+@CROSS_PRODUCT_INTRO@
 
+====================
+How to use this tool
+====================
 
-===========
-Description
-===========
+@GALAXY_DOT_PRODUCT_SEMANTICS@
 
+Running input lists through this tool produces new list structures (described in detail below) that when using
+the same natural element-wise matching "map" over semantics described above produce every combination of the
+elements of the two lists compared against each other. Running a tool with these two outputs instead of the inital
+two input produces a nested list structure where the jth element of the inner list of the ith element of the outer
+list is a comparison of the ith element of the first list to the jth element of the second list. 
+Put more simply, the result is a nested list where the identifiers of an element describe which inputs were
+matched to produce the comparison output found at that element. 
+
+.. image:: ${static_path}/images/tools/collection_ops/nested_crossproduct_output.png
+  :alt: The Cartesian Product of Two Collections
+  :width: 500
+
+============================================
+What this tool does (technical details)
+============================================
+
+This tool consumes two lists - we will call them ``input_a`` and ``input_b``. If ``input_a``
```

Comment:
Would call those input `A` and `B`. And the lists `a_1,...a_n` and `b_1,...,b_m`.

Reply:
@bernt-matthias The text doesn't refer to the inputs at all, I don't think we need to name them.

I would perhaps just say
```suggestion
This tool consumes two flat lists. We will call the input collections ``input_a`` and ``input_b``. If ``input_a``
```

Reply:
Then, why is it called `input_a`?

My main point would be not to use indices (with underscores) here (and instead use it for the list elements).

Reply:
It's my expectation from the more formal/mathematical papers I encountered. With underscores I expect elements of a list.

That is, my preference would be underscores for the lists and no underscores for the two inputs.

In the end I don't care that much. Formally it's well defined and fine. Just a bit more pleasant for me 😅